src/main/kotlin/xyz/ofortune/app/keycloak/RegistrationTurnstile.kt

```diff
@@ -92,7 +94,7 @@ class RegistrationTurnstile : FormAction, FormActionFactory, ConfiguredProvider
                 val content = response.entity.content
                 val json = JsonSerialization.readValue(content, Map::class.java)
                 LOGGER.tracef("Turnstile response: %s", json)
-                json["success"] == true && json["action"] == action
+                json["success"] == true && (captcha == TURNSTILE_DUMMY_TOKEN || json["action"] == action)
```

Comment:
While their docs suggest checking action, the testing token doesn't return action (presumably because they can't tie a dynamic action to a dummy static token).

![image](https://github.com/panpaul/keycloak-turnstile/assets/579018/55f7857a-9154-42e6-86ea-7f01803731be)

Thus, the action check as-is prevents testing with the dummy keys and secrets.

That said, if the dummy key is spoofed only on the client side (while using a live secret), the token will be rejected without the action check being necessary.
![image](https://github.com/panpaul/keycloak-turnstile/assets/579018/964f31eb-444e-4a10-a43f-9d7c658c9062)
![image](https://github.com/panpaul/keycloak-turnstile/assets/579018/eae4c40b-6fb8-44f9-900e-4604bd9978d8)